cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -431,9 +454,23 @@ func (m *McmManager) SetMachineDeploymentSize(machinedeployment *MachineDeployme
 	return nil
 }
 
+func (m *McmManager) getMachineDeploymentObjUntilDeadline(mdName string, retryDeadline time.Time) (*v1alpha1.MachineDeployment, error) {
```

Comment:
This code is not very clean. If there is an error you return nil just because you wish to retry in the calling method. So in a way the implementation of this method is completely tied to the caller.
Each function should independently make sense and should be complete. Splitting this logic into another methods seems artificial to me.